test/proxy-bcrypt.test.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const { createUnzip } = require('zlib');
+
+const tar = require('tar-fs');
+const Agent = require('https-proxy-agent');
+const fetch = require('node-fetch');
+const rimraf = require('rimraf');
+
+const test = require('tape');
+
+// this is a derived from build.test.js and should be kept in sync with it
+// as much as possible.
+const { mockS3Http } = require('../lib/node-pre-gyp.js');
+const proxy = require('./proxy.util');
+
+const proxyPort = 8124;
+const proxyServer = `http://localhost:${proxyPort}`;
+// options for fetch
+const options = {};
+
+let initial_s3_host;
+let initial_mock_s3;
+
+// the temporary download directory and file
+const downloadDir = 'download';
```

Comment:
How about using `os.tmpdir()` instead of a local directory (that would need to be put into .gitignore and .npmignore)?

Reply:
will do

Reply:
done